tuna/driver.py

```diff
@@ -0,0 +1,94 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Module that encapsulates the DB representation of a Driver cmd"""
+
+from typing import Union, Dict, Any
+from abc import ABC, abstractmethod
+from tuna.utils.logger import setup_logger
+
+LOGGER = setup_logger('driver_base')
+
+
+class DriverBase(ABC):
+  """Represents db tables based on ConfigType"""
+
+  @abstractmethod
+  def construct_driver(self, line: str) -> bool:
+    """Takes a MIOpenDriver cmd or PDB key"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def construct_driver_from_db(self, db_obj: Any) -> bool:
+    """Takes a <>_config row and returns a driver cmd"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def insert_tensor(self, tensor_dict: dict) -> int:
+    """Insert new row into tensor table and return primary key"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def get_input_t_id(self) -> int:
+    """Build 1 row in tensor table based on layout from fds param
+       Details are mapped in metadata LAYOUT"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def compose_input_t(self) -> Dict[str, int]:
+    """Build input_tensor"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def decompose_input_t(self, db_obj: Any) -> bool:
+    """Use input_tensor to assign local variables to build driver cmd """
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def compose_fds(self, tok: list, line: str) -> bool:
+    """Compose fds from driver line"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def get_weight_t_id(self) -> int:
+    """Build 1 row in tensor table based on layout from fds param
+     Details are mapped in metadata LAYOUT"""
+    raise NotImplementedError("Not implemented")
+
+  @abstractmethod
+  def parse_driver_line(self, line: str):
```

Comment:
This may be the same as construct_driver in the general case.

Reply:
just a heads up that if you rename this, there will be some code changes in other files

Reply:
Let's remove parse_driver_line from the interface as it is essentially a sub-method of construct_driver.